main/api/src/mill/api/SystemStreams.scala

```diff
@@ -7,3 +7,30 @@ class SystemStreams(
     val err: PrintStream,
     val in: InputStream
 )
+
+object SystemStreams{
+  val original = new SystemStreams(System.out, System.err, System.in)
```

Comment:
I don't think this works reliable, as objects are only instantiated when their class is loaded/accessed (I think). It works for us, as we use `SystemStream` from `MillMain`, which is "early enough", but there is no guarantee this still works, if we change `MillMain` later.

Either we set this from outside, or we add some notes about these potential issue.

Reply:
So my idea was that this is guaranteed to work on the condition that we only use `SystemStreams.withStreams` to do the redirects. I updated all the existing code to do so, and is probably a best practice anyway in order to keep the logic consistent throughout the codebase.

Now I can't guarantee that someone in future won't add code that redirects stuff manually using `System.setOut` or `Console.withOut`, but as long as we stick with the convention of using `SystemStreams.withStreams`, then this should work robustly